src/Nixfmt/Pretty.hs

```diff
@@ -536,6 +540,12 @@ isAbsorbable (Path _) = True
 -- Non-empty sets and lists
 isAbsorbable (Set _ _ (Items (_ : _)) _) = True
 isAbsorbable (List _ (Items (_ : _)) _) = True
+-- Empty sets and lists if they have a line break
+-- https://github.com/NixOS/nixfmt/issues/253
+isAbsorbable (Set _ (Ann{sourceLine = line1}) (Items []) (Ann{sourceLine = line2}))
+  | line1 /= line2 = True
+isAbsorbable (List (Ann{sourceLine = line1}) (Items []) (Ann{sourceLine = line2}))
+  | line1 /= line2 = True
```

Comment:
```suggestion
isAbsorbable (Set _ (Ann{sourceLine = line1}) (Items []) (Ann{sourceLine = line2})) = line1 /= line2
isAbsorbable (List (Ann{sourceLine = line1}) (Items []) (Ann{sourceLine = line2})) = line1 /= line2
```

I think this can be simplified a bit

Reply:
Hm, while semantically equivalent, it has a slightly different intent and would break if someone added a slightly overlapping case later

Reply:
Yeah it makes sense, feel free to ignore